opm/parser/eclipse/Deck/Deck.hpp

```diff
@@ -113,17 +115,24 @@ namespace Opm {
             const UnitSystem& getDefaultUnitSystem() const;
             const UnitSystem& getActiveUnitSystem() const;
 
+            const std::string getDataFile() const;
+            void setDataFile(const std::string& dataFile);
+            bool hasDataFile() const;
 
         private:
             Deck( std::vector< DeckKeyword >&& );
 
             void initUnitSystem() const;
 
+
             std::vector< DeckKeyword > keywordList;
             mutable MessageContainer m_messageContainer;
 
             mutable std::unique_ptr< UnitSystem > activeUnits;
             mutable std::unique_ptr< UnitSystem > defaultUnits;
+
+            std::string m_dataFile;
+            bool m_hasDataFile;
```

Comment:
1. do we need bool and string, or can you have empty string meaning no data file?
2. what is the intention of `m_dataFile`?  Can you provide a comment, as in, if I set dataFile to be a different file than what the deck was constructed from, can I expect anything to happen except that dataFile is now a different path?
3.  What if I call setDataFile to a something which is not a valid POSIX path, or to somewhere I don't have read/write access?